deploy/pivot_role/pivotRole.yaml

```diff
@@ -439,12 +439,25 @@ Resources:
               - 'iam:ListRoles'
             Effect: Allow
             Resource: '*'
-          - Sid: IAMRolePolicy
+          - Sid: IAMRolePolicy1
             Action:
               - 'iam:PutRolePolicy'
               - 'iam:DeleteRolePolicy'
+              - 'iam:AttachRolePolicy'
+              - 'iam:DetachRolePolicy'
+              - 'iam:ListAttachedRolePolicies'
             Effect: Allow
-            Resource: '*'
+            Resource:
+              - !Sub 'arn:aws:iam::${AWS::AccountId}:role/*'
+          - Sid: IAMRolePolicy2
+            Action:
+              - 'iam:CreatePolicy'
+              - 'iam:DeletePolicy'
+              - 'iam:CreatePolicyVersion'
+              - 'iam:DeletePolicyVersion'
+            Effect: Allow
+            Resource:
+              - !Sub 'arn:aws:iam::${AWS::AccountId}:policy/${EnvironmentResourcePrefix}*'
```

Comment:
same comment as data sharing pivot role policy to include delete policy permissions for the inline policies we are removing,
`dataall-targetDatasetS3Bucket-AccessControlPolicy` and `targetDatasetAccessControlPolicy`

Reply:
Confirmed: ` Failed to delete policy targetDatasetAccessControlPolicy of role mar-science-march-a2-xbvhdms8 : An error occurred (NoSuchEntity) when calling the DeleteRolePolicy operation: The role policy with name targetDatasetAccessControlPolicy cannot be found.
cdkproxy-1       | Failed to delete policy dataall-targetDatasetS3Bucket-AccessControlPolicy of role mar-science-march-a2-xbvhdms8 : An error occurred (NoSuchEntity) when calling the DeleteRolePolicy operation: The role policy with name dataall-targetDatasetS3Bucket-AccessControlPolicy cannot be found.
` I'll fx it